src/actions.sol

```diff
@@ -57,9 +71,41 @@ contract Actions {
     event ApproveNFT(address indexed registry, address indexed usr, uint256 tokenAmount);
     event ApproveERC20(address indexed erc20, address indexed usr, uint256 amount);
     event TransferERC20(address indexed erc20, address indexed dst, uint256 amount);
+    event Minted(address indexed registry, uint256 tokenId);
+
+    address public immutable shelf;
+    address public immutable pile;
+    address public immutable feed;
```

Comment:
```suggestion
    address public shelf;
    address public pile;
    address public feed;
```

Setting these three vars as immutable throws the error:
`TypeError: Immutable variables cannot be read during contract creation time, which means they cannot be read in the constructor or any function or modifier called from it.`

Reply:
They need to be `immutable`. You can't store them in `storage` in a logic contract in a delegate call pattern.
It will access the storage of the proxy at slot of `shelf`, `pile` and `feed`. 

Interesting, that I didn't get this error in the test. 
I guess it is because of the `address(0x0)` check. I think this can be fixed with a local variable.